tests/test_equivalence.py

```diff
@@ -20,13 +21,43 @@
 from fasm2bels.fasm2bels import main
 
 
-test_names = ["add32"]
+test_names = ["add32","alu"]
 
 
 def unpack_tar(tar_file):
     tar = tarfile.open(name=tar_file, mode="r:gz")
     tar.extractall(path=os.path.dirname(tar_file))
 
+def createGoldenfile(fileName):
+    path = os.environ['VIVADO_PATH'] #Note: Please use export VIVADO_PATH=/tools/Xilinx/Vivado/(VERSION OF VIVADO YOU'RE USING)/bin/vivado before running this.
+    file_exists = exists("equivalence_checking_data/"+fileName+"/top_bit.golden.v")
+    if(file_exists):
+        os.system("echo GoldenFile already exists.")
+        toTCL = "rm -r __pycache__"
+        os.system(toTCL)
+    else:
+        file_exists = exists(path)
+        if(file_exists):
+            updateTCL('FILENAME', fileName)
+            
+            os.system(path +  " -mode batch -source createGoldenFile.tcl")
+        
+            updateTCL(fileName, 'FILENAME')
+            
+            toTCL = "rm vivado* && rm -r *.dcp && rm -r *.edf && rm report_io.txt && rm -r .Xil && mv *_impl.v equivalence_checking_data/" + fileName + "/top_bit.golden.v"
+            os.system(toTCL)
+
+        else:
+            os.system("echo GoldenFile cannot be generated. Vivado must be installed in order for goldenfile to be created.")
+
+def updateTCL(findStr, replaceStr):
+    with open('createGoldenFile.tcl', 'r') as file:
+        filedata = file.read()
+    
+    filedata = filedata.replace(findStr, replaceStr)
+
+    with open('createGoldenFile.tcl', 'w') as file:
+        file.write(filedata)
```

Comment:
This can be removed when the argument parsing is added to the tcl script. The tcl script is basically a template that will be used by all the tests, and shouldn't be updated.

E.g. think if there is parallel execution of the tests. In that case, updating the tcl file would definitely create conflicts.